### ISSUE TITLE
Cannot get any detection and evaluation gives zero mAP also

### ISSUE THREAD

**Issue by eypros** [pierluigiferrari/ssd_keras#267]
I was just trying to verify the code works as expected so I tried to retrain the SSD300 on Pascal VOC. I followed the provided [instructions](https://github.com/pierluigiferrari/ssd_keras/blob/master/ssd300_training.ipynb) for performing the training and as I checked the loss was consistent with the expected one in the [training summary](https://github.com/pierluigiferrari/ssd_keras/blob/master/training_summaries/ssd300_pascal_07+12_training_summary.md).

I didn't finish my training but I tried to verify that everything was fine by applying the [evaluation script](https://github.com/pierluigiferrari/ssd_keras/blob/master/ssd300_evaluation.ipynb) provided. I applied the evaluation script after epoch 62. My training loss and validation loss was:
`61,4.6116460539566075,4.67574555423225` which seemed consistent with the plot provided.
With surprise I noticed that my mAP is zero (0.0) for all classes. To be exact I had to modify the [code](https://github.com/JeroenDelcour/ssd_keras/blob/master/eval_utils/average_precision_evaluator.py#L616) (according to these [instructions](https://github.com/pierluigiferrari/ssd_keras/issues/232)) to even get these results since a bug is present if no detection are present. Anyway this not the problem now.

The problem is that I am not getting any predictions what so ever from my model. I couldn't even get any prediction from the provided "Trained from Scratch" (the one with mAP 0.771) model either, so, I guess the problem is not with my model.

Can someone provide any clue of what's the problem with my case. I didn't make any changes in the code except for the batch size (I reduced it to 30 from 32 to be sure it will not get OOM issues) and converted the code to pure python (not notebook). I have tried `ssd300_inference.py` as well as `ssd300_evaluation.py` and they are consistent to not produce anything useful. I guess I have made a huge mistake somewhere because this does not make any sense.

**Comment by liguilan1227** [comment]
the same issue:
ons for class 2/20
No predictions for class 3/20
No predictions for class 4/20
No predictions for class 5/20
No predictions for class 6/20
No predictions for class 7/20
No predictions for class 8/20
No predictions for class 9/20
No predictions for class 10/20
No predictions for class 11/20
No predictions for class 12/20
No predictions for class 13/20
No predictions for class 14/20
No predictions for class 15/20
No predictions for class 16/20
No predictions for class 17/20
No predictions for class 18/20
No predictions for class 19/20
No predictions for class 20/20
Computing precisions and recalls, class 1/20
Computing precisions and recalls, class 2/20
Computing precisions and recalls, class 3/20
Computing precisions and recalls, class 4/20
Computing precisions and recalls, class 5/20
Computing precisions and recalls, class 6/20
Computing precisions and recalls, class 7/20
Computing precisions and recalls, class 8/20
Computing precisions and recalls, class 9/20
Computing precisions and recalls, class 10/20
Computing precisions and recalls, class 11/20
Computing precisions and recalls, class 12/20
Computing precisions and recalls, class 13/20
Computing precisions and recalls, class 14/20
Computing precisions and recalls, class 15/20
Computing precisions and recalls, class 16/20
Computing precisions and recalls, class 17/20
Computing precisions and recalls, class 18/20
Computing precisions and recalls, class 19/20
Computing precisions and recalls, class 20/20
Computing average precision, class 1/20
Computing average precision, class 2/20
Computing average precision, class 3/20
Computing average precision, class 4/20
Computing average precision, class 5/20
Computing average precision, class 6/20
Computing average precision, class 7/20
Computing average precision, class 8/20
Computing average precision, class 9/20
Computing average precision, class 10/20
Computing average precision, class 11/20
Computing average precision, class 12/20
Computing average precision, class 13/20
Computing average precision, class 14/20
Computing average precision, class 15/20
Computing average precision, class 16/20
Computing average precision, class 17/20
Computing average precision, class 18/20
Computing average precision, class 19/20
Computing average precision, class 20/20
aeroplane     AP    0.0
bicycle       AP    0.0
bird          AP    0.0
boat          AP    0.0
bottle        AP    0.0
bus           AP    0.0
car           AP    0.0
cat           AP    0.0
chair         AP    0.0
cow           AP    0.0
diningtable   AP    0.0
dog           AP    0.0
horse         AP    0.0
motorbike     AP    0.0
person        AP    0.0
pottedplant   AP    0.0
sheep         AP    0.0
sofa          AP    0.0
train         AP    0.0
tvmonitor     AP    0.0

              mAP   0.0
Matching predictions to ground truth, class 1/20.:  23%|鈻堚枅鈻堚枅鈻堚枅鈻堚枅鈻�                           | 5179/22337 [00:00<00Matching predictions to ground truth, class 1/20.:  53%|鈻堚枅鈻堚枅鈻堚枅鈻堚枅鈻堚枅鈻堚枅鈻堚枅鈻堚枅鈻堚枅鈻�                | 1Matching predictions to ground truth, class 1/20.:  86%|鈻堚枅鈻堚枅鈻堚枅鈻堚枅鈻堚枅鈻堚枅鈻堚枅鈻堚枅鈻堚枅鈻堚枅鈻堚枅鈻堚枅鈻堚Matching predictions to ground truth, class 1/20.: 100%|鈻堚枅鈻堚枅鈻堚枅鈻堚枅鈻堚枅鈻堚枅鈻堚枅鈻堚枅鈻堚枅鈻堚枅鈻堚枅鈻堚枅鈻堚00, 65187.98it/s]
No predictions for class 2/20
No predictions for class 3/20
No predictions for class 4/20
No predictions for class 5/20
No predictions for class 6/20
No predictions for class 7/20
No predictions for class 8/20
No predictions for class 9/20
No predictions for class 10/20
No predictions for class 11/20
No predictions for class 12/20
No predictions for class 13/20
No predictions for class 14/20
No predictions for class 15/20
No predictions for class 16/20
No predictions for class 17/20
No predictions for class 18/20
No predictions for class 19/20
No predictions for class 20/20
Computing precisions and recalls, class 1/20
Computing precisions and recalls, class 2/20
Computing precisions and recalls, class 3/20
Computing precisions and recalls, class 4/20
Computing precisions and recalls, class 5/20
Computing precisions and recalls, class 6/20
Computing precisions and recalls, class 7/20
Computing precisions and recalls, class 8/20
Computing precisions and recalls, class 9/20
Computing precisions and recalls, class 10/20
Computing precisions and recalls, class 11/20
Computing precisions and recalls, class 12/20
Computing precisions and recalls, class 13/20
Computing precisions and recalls, class 14/20
Computing precisions and recalls, class 15/20
Computing precisions and recalls, class 16/20
Computing precisions and recalls, class 17/20
Computing precisions and recalls, class 18/20
Computing precisions and recalls, class 19/20
Computing precisions and recalls, class 20/20
Computing average precision, class 1/20
Computing average precision, class 2/20
Computing average precision, class 3/20
Computing average precision, class 4/20
Computing average precision, class 5/20
Computing average precision, class 6/20
Computing average precision, class 7/20
Computing average precision, class 8/20
Computing average precision, class 9/20
Computing average precision, class 10/20
Computing average precision, class 11/20
Computing average precision, class 12/20
Computing average precision, class 13/20
Computing average precision, class 14/20
Computing average precision, class 15/20
Computing average precision, class 16/20
Computing average precision, class 17/20
Computing average precision, class 18/20
Computing average precision, class 19/20
Computing average precision, class 20/20
aeroplane     AP    0.0
bicycle       AP    0.0
bird          AP    0.0
boat          AP    0.0
bottle        AP    0.0
bus           AP    0.0
car           AP    0.0
cat           AP    0.0
chair         AP    0.0
cow           AP    0.0
diningtable   AP    0.0
dog           AP    0.0
horse         AP    0.0
motorbike     AP    0.0
person        AP    0.0
pottedplant   AP    0.0
sheep         AP    0.0
sofa          AP    0.0
train         AP    0.0
tvmonitor     AP    0.0

              mAP   0.0
 how to solve it?
